config/v1/types_console.go

```diff
@@ -26,6 +28,31 @@ type Console struct {
 type ConsoleSpec struct {
 	// +optional
 	Authentication ConsoleAuthentication `json:"authentication"`
+	// route contains hostname and serving certificate for replacing the default route for the console.
+	// If a custom route is specified, a new route will be created with the provided hostname.
+	// The default console route will be maintained to reserve the default hostname for console if the custom route is removed.
+	// If not specified, default route will be 	used.
+	// +optional
+	Route ConsoleConfigRoute `json:"route,omitempty"`
+}
+
+// ConsoleConfigRoute holds information on external route access to console.
+type ConsoleConfigRoute struct {
+	// name of the route to be created.
+	Name string `json:"name"`
+	// hostname is the desired custom address under which console will be available.
+	// +optional
+	Hostname string `json:"hostname,omitempty"`
+	// secretName points to secret in the openshift-config namespace that
+	// contains custom certificate and key and needs to be created manually by
+	// the cluster admin.
+	// Referenced Secret is required to contain following key value pairs:
+	// - "tls.crt" - to specifies custom certificate
+	// - "tls.key" - to specifies private key of the custom certificate
+	// If hostname that uses default routing suffix of the given cluster, the
+	// servignCert specification will not be needed.
+	// +optional
+	SecretName string `json:"secretName,omitempty"`
```

Comment:
Guessing this should really be a `SecretNameReference` rather than a `string`.
Not a lot to it:

```
// SecretNameReference references a secret in a specific namespace.
// The namespace must be specified at the point of use.
type SecretNameReference struct {
	// name is the metadata.name of the referenced secret
	// +kubebuilder:validation:Required
	// +required
	Name string `json:"name"`
}
```